moto/glue/models.py

```diff
@@ -375,8 +403,173 @@ def register_schema_version(self, schema_id, schema_definition):
+                get_schema_version_dict["DataFormat"] = schema.data_format
+                return get_schema_version_dict
+        raise SchemaVersionNotFoundFromSchemaIdException(
+            registry_name, schema_name, schema_arn, version_number, latest_version
+        )
+
+    def get_schema_by_definition(self, schema_id, schema_definition):
+        """GetSchemaByDefinition API"""
+        # Validate SchemaId
+        validate_schema_definition_length(schema_definition)
+        registry_name, schema_name, schema_arn = validate_schema_id(
+            schema_id, self.registries
+        )
+
+        # Get Schema By Definition
+        schema = self.registries[registry_name].schemas[schema_name]
+        for schema_version in schema.schema_versions.values():
+            if (
+                schema_definition == schema_version.schema_definition
+                and schema_version.schema_version_status != DELETING_STATUS
+            ):
+                get_schema_by_definition_dict = (
+                    schema_version.get_schema_by_definition_as_dict()
+                )
+                get_schema_by_definition_dict["DataFormat"] = schema.data_format
+                return get_schema_by_definition_dict
+        raise SchemaNotFoundException(schema_name, registry_name, schema_arn)
+
+    def put_schema_version_metadata(
+        self, schema_id, schema_version_number, schema_version_id, metadata_key_value
+    ):
+        """PutSchemaVersionMetadata API"""
+        # Validate metadata_key_value and schema version params
+        (
+            metadata_key,
+            metadata_value,
+        ) = validate_schema_version_metadata_pattern_and_length(metadata_key_value)
+        (
+            schema_version_id,
+            registry_name,
+            schema_name,
+            schema_arn,
+            version_number,
+            latest_version,
+        ) = validate_schema_version_params(
+            self.registries, schema_id, schema_version_id, schema_version_number
+        )
+
+        # PutSchemaVersionMetadata using SchemaVersionId
+        if schema_version_id:
+            for registry in self.registries.values():
+                for schema in registry.schemas.values():
+                    if schema.schema_versions.get(schema_version_id, None):
+                        metadata = schema.schema_versions[schema_version_id].metadata
+                        validate_number_of_schema_version_metadata_allowed(metadata)
+
+                        if metadata_key in metadata:
+                            if metadata_value in metadata[metadata_key]:
+                                raise SchemaVersionMetadataAlreadyExistsException(
+                                    schema_version_id, metadata_key, metadata_value
+                                )
+                            metadata[metadata_key].append(metadata_value)
+
+                        metadata[metadata_key] = [metadata_value]
```

Comment:
```suggestion
                            metadata[metadata_key].append(metadata_value)
                        else:
                            metadata[metadata_key] = [metadata_value]
```
Now that it's more readable - it's suddenly looks like there's an `else` missing :)